File: dk09.cpp
/*Hailun*/
#include "iostream"
using namespace std;
bool isLeapYear(int year) 
{
    if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0))
        return true;
    return false;
}

int main()
{
    int year;
    cin >> year;
    if(year <= 0 || year > 100000)
    {
        cout << "INVALID" ;
        exit(0);
    }
    if(isLeapYear(year))    cout << "YES" ;
    else    cout << "NO" ;
    return 0;
}

File: dk09.java
package luyencode;

import java.util.Scanner;

public class dk09 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int year = sc.nextInt();
        if (year <= 0 || year > 100000) {
            System.out.println("INVALID");
        } else if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
        sc.close();
    }
}
